edd-hide-download.php

```diff
@@ -40,6 +40,7 @@ final class EDD_Hide_Download {
 		 * Keep the hidden downloads in options
 		 *
 		 * @since  1.2
+		 * @since  1.2.11 Deprecated
 		 * @var    array
 		 */
 		private $hidden_downloads;
```

Comment:
```suggestion
		private $hidden_downloads = array();
```
Would this not address the issue in a slightly different way? Since we are still setting this up in the `hooks` method even though it's not used, at least calling it should always result in an array at the very least.

Reply:
While it would stop the symptom (that the property isn't properly set to a value that can have `array_merge` run on it), it doesn't fix the problem which is that the actual method to set that isn't being run due to either a load order or a class instantiation not being in a 'singleton' pattern.